rules/hardcoded_credentials.go

```diff
@@ -127,6 +167,7 @@ func (r *credentials) matchEqualityCheck(binaryExpr *ast.BinaryExpr, ctx *gosec.
 // assigned to variables that appear to be related to credentials.
 func NewHardcodedCredentials(id string, conf gosec.Config) (gosec.Rule, []ast.Node) {
 	pattern := `(?i)passwd|pass|password|pwd|secret|token|pw|apiKey|bearer|cred`
+	patternValue := "^(?i)[a-f0-9]{64}$"
```

Comment:
Many API keys also use `-_` in their syntax. Also, the length of 64 seems arbitrary to me. [Talisman](https://github.com/thoughtworks/talisman) is a secret scanner written in go, they may have a good solution that can be mimicked here

Reply:
Yes you are right, it was arbitrary. I just put one in to specifically get VirusTotal API keys. I am taking a look at the repo you linked. I will do some digging and testing to figure out a good solution.

Reply:
Yes you are right, it was arbitrary. I just put one in to specifically get VirusTotal API keys. I am taking a look at the repo you linked. I will do some digging and testing to figure out a good solution.

Reply:
Talisman is using various [regexes for their detector](https://github.com/thoughtworks/talisman/blob/e8b8eba8718b137d723007877f3d105498193a81/detector/pattern/pattern_detector.go#L5]); I am not sure if there is a more general solution that would catch a wide variety of API keys.  So I found some regex patterns for a couple of other keys [here](https://github.com/odomojuli/RegExAPI). I added regex for Google API keys, GitHub Fine-Grained PATs, and Google OAuth keys, as well as tests for each. The tests are run against some fake API keys I randomly generated based on the pattern.